sympy/ntheory/tests/test_generate.py

```diff
@@ -107,7 +107,17 @@ def test_generate():
     assert nextprime(10**40) == (10**40 + 121)
     assert prevprime(97) == 89
     assert prevprime(10**40) == (10**40 - 17)
+
     assert list(sieve.primerange(10, 1)) == []
+    assert list(sieve.totientrange(5, 15)) == [4, 2, 6, 4, 6, 4, 10, 4, 12, 6]
+    assert list(sieve.totientrange(900, 1000)) == [totient(x) for x in range(900, 1000)]
+    assert list(sieve.totientrange(0, 1)) == []
+    assert list(sieve.totientrange(1, 2)) == [1]
+    assert list(sieve.mobiusrange(5, 15)) == [-1, 1, -1, 0, 0, 1, -1, 0, -1, 1]
+    assert list(sieve.mobiusrange(1050, 1100)) == [mobius(x) for x in range(1050, 1100)]
```

Comment:
Line 113 and 117 tests pass.
But these tests wouldn't mean anything if `totient()` and `mobius()` are used to return values (possibly later) which have already been `sieved`.